plugins/service.js

```diff
@@ -0,0 +1,59 @@
+const Plugin = require('./base');
+const { Scopes } = require('../lib/constants');
+
+const setContext = (properties, service) => {
+    const merge = (acc, [key, to]) =>
+        !properties[to] && service[key] ?
+            Object.assign(acc, { [to]: service[key] }) : acc;
+    return Object.entries({
+        'name': 'service.name',
+        'version': 'service.version',
+        'instanceId': 'service.instance.id',
+        'namespace': 'service.namespace'
+    }).reduce(merge, properties);
+};
+
+const associateContext = (resource, service) => (connect) =>
+    (url, { clientProperties = {}, ...socketOptions } = {}) => {
+        if (!resource) {
+            clientProperties = setContext(clientProperties, service);
+        }
+        return connect(url, {
+            clientProperties,
+            ...socketOptions
+        });
+    };
+
+const queueName = (binding, exchange, service) => {
+    let name = `${service.name}:`;
+
+    if (typeof binding === 'string') name += binding;
+    else name += JSON.stringify(binding || null);
```

Comment:
How did you come up with this line? Binding is allowed to be something else than a string?

Reply:
This is for `headers` exchanges. We accept objects as bindings for them.

Reply:
Thanks! Missed that. In that case I suggest we use something like this

```js
Object.entries(binding).map(v => `${v[0]}=${v[1]}`).join(";")
// produces 
"format=pdf;pages=1"
```
just to save some bytes and have a better readability than json

Reply:
Thanks for the suggestion!